src/dist.rs

```diff
@@ -0,0 +1,321 @@
+use crate::{
+    anyhow::{ensure, Context, Result},
+    camino, clap, default_build_command, metadata,
+};
+use lazy_static::lazy_static;
+use std::{fs, path::PathBuf, process};
+use wasm_bindgen_cli_support::Bindgen;
+
+/// A helper to generate the distributed package
+///
+/// This struct provide a customizable way to assemble and generate a package
+/// for Wasm.
+///
+/// # Usage
+///
+/// ```rust,no_run
+/// use std::process;
+/// use xtask_wasm::{anyhow::Result, clap};
+///
+/// #[derive(clap::Parser)]
+/// enum Opt {
+///     Dist(xtask_wasm::Dist),
+/// }
+///
+/// fn main() -> Result<()> {
+///     let opt: Opt = clap::Parser::parse();
+///
+///     match opt {
+///         Opt::Dist(dist) => {
+///             let dist = dist
+///                 .static_dir_path("myproject/static")
+///                 .app_name("myproject")
+///                 .run_in_workspace(true)
+///                 .run("project")?;
+///
+///             log::info!("Built at {}", dist.dist_dir.display());
+///         }
+///     }
+///
+///     Ok(())
+/// }
+/// ```
+///
+/// Add a `dist` subcommand to assemble the distributed package for the
+/// `myproject` crate.
+///
+/// It will run the [`default_build_command`] at the workspace root, generate JS
+/// bindings and output two files: `project.js` and `project.wasm`. These two
+/// files will be written, and the content of the `project/static` directory
```

Comment:
It outputs 2 files and they will be written xD repetition again
```suggestion
/// bindings and output two files: `project.js` and `project.wasm`. The content of the `project/static` directory
```